web3.js/src/connection.ts

```diff
@@ -2554,17 +2601,33 @@ export class Connection {
     startSlot: number,
     endSlot: number,
   ): Promise<Array<TransactionSignature>> {
-    const unsafeRes = await this._rpcRequest(
-      'getConfirmedSignaturesForAddress',
-      [address.toBase58(), startSlot, endSlot],
-    );
-    const res = create(unsafeRes, GetConfirmedSignaturesForAddressRpcResult);
-    if ('error' in res) {
-      throw new Error(
-        'failed to get confirmed signatures for address: ' + res.error.message,
-      );
+    let options: any = {};
+
+    if (startSlot > 1) {
+      let start = startSlot - 1;
+      let startSignatures: Array<string> = [];
+      while (startSignatures.length == 0) {
+        const block = await this.getConfirmedBlockSignatures(start);
+        startSignatures = block.signatures;
+        start++;
```

Comment:
Should be searching back further right? If `start` goes to 0 then we can probably omit `until`?

Reply:
Ugh, yes, that was my intention. Thanks for catching!